packages/twenty-server/src/modules/workflow/workflow-executor/workspace-services/workflow-executor.workspace-service.ts

```diff
@@ -39,59 +40,64 @@ export class WorkflowExecutorWorkspaceService {
 
     const workflowAction = this.workflowActionFactory.get(step.type);
 
+    const inferredStepInput = await resolve(step.settings.input, context);
+
     const result = await workflowAction.execute({
-      step,
-      payload,
+      step: {
+        ...step,
+        settings: {
+          ...step.settings,
+          input: inferredStepInput,
+        },
+      },
+      context,
```

Comment:
I don't understand how it works. It looks like there is some inference happening here and in the workflow action (like in send-email.workflow-action.ts). I think it would be better to handle the inference in a single place, and I would suggest doing it here.